src/helpers/ParseQuery.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * Returns what the input is (zip code, county name, or address)
+ * @Param str The string to parse
+ * @Returns A string telling what the input is
+ */
+export default function Parse(str: string): string {
+  const zip = new RegExp('^\\d{5}$');
+  // let county = new RegExp('(.+), ([a-z]{2})');
+  // let county: RegExp = /(.+), ([a-z]{2})/;
+  const county = new RegExp('(^[A-z]+), ([A-z]{2})');
+  const add1 = new RegExp('(.+), ([A-z]{2})');
+  // let add2: RegExp = /\d{1,6}\s(?:[A-Za-z0-9#]+\s){0,7}(?:[A-Za-z0-9#]+,)\s*(?:[A-Za-z]+\s){0,3}(?:[A-Za-z]+,)\s*[A-Z]{2}\s*\d{5}/;
+
+  console.log('zipcode: ', zip.test(str));
+  console.log('county: ', county.test(str));
+  console.log('add1: ', add1.test(str));
+  // console.log("add2: ", add2.test(str));
+
+  if (zip.test(str)) {
+    return 'zip';
+  }
+  if (county.test(str)) {
+    return 'county';
+  }
+  if (add1.test(str)) {
+    return 'add1';
+  }
+  // if (add2.test(str)){
+  //    return 'add2';
+  // }
+
+  return 'error';
```

Comment:
It definitely works very well. I just have a suggestion that will make it a bit more readable (at least to me), scalable, and condense.
```suggestion
  return zip.test(str) && 'zip'
    || county.test(str) && 'county'
    || add1.test(str) && 'add1'
    || 'error'
```

Reply:
Hmm it's definitely more concise and elegant, but I think how it's written before is a bit more readable since it's simpler.  I'm not sure it is more scalable though. Doesn't each part in a boolean expression still have to get executed in order to take advantage of the short-circuiting?

Reply:
I am not a sure, but I think that it breaks as soon as one && statement is reached. It's probably only the preference at this point. I don't believe there is one right way to do it. I think you should do whatever you think will work best for our team.

Reply:
The before version seems much more clearer on what the conditions are but I feel like @huncfut version seems to make it clear that you are returning only one value and that value is dependent on the different conditions.

Reply:
I also think it should be consistent throughout the team. It would be hard to follow if one person used if statements and someone else used the concise version.

Reply:
I think having the if statements is a bit more readable in this scenario but the other way would work the same